weather_dl/download_pipeline/parsers.py

```diff
@@ -159,11 +158,11 @@ def parse_config(file: t.IO) -> Config:
         config.read_file(file)
         config_by_section = {s: _parse_lists(config, s) for s in config.sections()}
         config_with_nesting = parse_subsections(config_by_section)
-        return config_with_nesting
+        return Config.from_config(config_with_nesting)
     except configparser.ParsingError:
         pass
 
-    return {}
+    return Config.from_config({})
```

Comment:
Sorry to suggest this, because it may involve a lot of undoing, but: I think that it may be better to create a config object at the end of `process_config`, once all the validation steps have finished. I can think of two reasons why this makes sense to me; please let me know if you agree: 
1. The `parse_config` function is kind of our "lexing" & "parsing" phase ["Map of the territory", _[Crafting Interpreters]_]. It parses an abstract-syntax tree (in our case, JSON-like Python-Dicts) from raw text. Ideally, we shouldn't mix parsing and analysis. There are some advantages for keeping those steps separate; I see them as separate concerns. (In this analogy, I see the `Config` dataclass as an "Intermediate Representation".)
2. I argue it's a bit more efficient to check for syntax errors before creating the dataclass, which requires iterating through the entire nested object. 

What do you think about this? What are your thoughts about transforming to the dataclass earlier? 

[Crafting Interpreters]: https://craftinginterpreters.com/a-map-of-the-territory.html

Reply:
Yes, it makes sense. We should create dataclass object at the end of `process_config`.